pkg/services/accesscontrol/middleware/middleware.go

```diff
@@ -46,6 +47,12 @@ func Deny(c *models.ReqContext, evaluator accesscontrol.Evaluator, err error) {
 		)
 	}
 
+	if !c.IsApiRequest() {
+		// TODO(emil): I'd like to show a message after this redirect, not sure how that can be done?
```

Comment:
I think you could pass an id that can be used to look up an error message in a hard coded list.